### PR TITLE
Fire skipped events at debug level

### LINKED ISSUE #8774: [CT-3185] Hundreds of: Skipping due to fail_fast
### Is this a new bug in dbt-core?

- [X] I believe this is a new bug in dbt-core
- [X] I have searched the existing issues, and I could not find an existing issue for this bug

I searched open and closed issues for "Skipping due to fail_fast".

### Current Behavior

I can get as many as hundreds of these. 900 in the run I've just done, because something near the beginning of the dag failed - because I'm setting up clones of databases with different permissions to test something.

This wasn't always the case, I feel like it started fairly recently. Weeks ago, not months ago. We moved from 1.5.2 to 1.6.1 on the 13th of September. I checked the behaviour is the same in 1.6.1 and 1.6.5.

```
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
00:46:42    Skipping due to fail_fast
00:46:42
```

### Expected Behavior

Since it is not saying what it is skipping, and all of those are the same, it should just say something once. Not sure what. Maybe "Skipping ### models due to fail_fast", ### being the number of models skipped.

### Steps To Reproduce

1. Set up a failure at the beginning of the DAG.
2. Run: `dbt run --fail-fast`

### Relevant log output

_No response_

### Environment

```markdown
- OS: Linux
- Python: 3.10.13
- dbt: 1.6.5
- dbt-snowflake: 1.6.4
```

### Which database adapter are you using with dbt?

snowflake

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by nevdelap**
Resolves #8774. Picking up from Scott Gigantes #9371. Scott's not had the time to finish it off and has suggested I take a look. 

### TODO

* Set up my dev env to reproduce the issue using this clone locally installed, and see it fixed with the patch. :heavy_check_mark: 
* Learn about contributing to this project. :heavy_check_mark: 
* Resolve the issues described in the [PR](https://github.com/dbt-labs/dbt-core/pull/9371#pullrequestreview-1980769778). :heavy_check_mark: 

### Problem
If fail_fast, dbt prints hundreds of lines "Skipping due to fail_fast", hiding the actual cause of the error deep in bash history (sometimes too deep to read)

### Solution
This moves the "Skipping due to fail_fast" to log level DEBUG. An alternative would be to a) drop it altogether or b) aggregate the results together to a "Skipping tasks due to fail_fast". This approach requires the least modification to the task results framework.

### Checklist

- [x] I have read [the contributing guide](https://github.com/dbt-labs/dbt-core/blob/main/CONTRIBUTING.md) and understand what's expected of me  
- [x] I have run this code in development and it appears to resolve the stated issue  
- [x] This PR includes tests, or tests are not required/relevant for this PR
- [x] This PR has no interface changes (e.g. macros, cli, logs, json artifacts, config files, adapter interface, etc) or this PR has already received feedback and approval from Product or DX
- [x] This PR includes [type annotations](https://docs.python.org/3/library/typing.html) for new and modified functions

### CODE REVIEW

**Comment by QMalcolm** [comment]
@nevdelap Let me know if you need any guidance or want to pair to get this across the line 🙂

**Comment by nevdelap** [comment]
Thanks @QMalcolm. I'll expect to do some more on Monday. I see your link to where the tests are likely to go. If I have any trouble I'll ping you. :+1:

**Comment by QMalcolm** [comment]
Thanks for signing the CLA - @cla-bot check

**Comment by QMalcolm** [comment]
### For documentation purposes:

As mentioned on the PR by @scottgigante (#9371), what we need to do before merging this is add a test for the added functionality to ensure we don't regress in the future.

In the test(s) we should:
- assert that Skipping due to fail_fast doesn't appear in the output if --fail-fast is specified but --debug is not
- assert that Skipping due to fail_fast does appear in the ouput if both --fail-fast and --debug are specified


Here are [some examples](https://github.com/dbt-labs/dbt-core/blob/a994ace2dbc54ee3b355cca1fcf4ba3c9c106edd/tests/functional/minimal_cli/test_minimal_cli.py) of tests that something exists in the run output. Additionally a good place for the test would be in [test_fail_fast_run.py](https://github.com/dbt-labs/dbt-core/blob/.a994ace2dbc54ee3b355cca1fcf4ba3c9c106edd/tests/functional/fail_fast/test_fail_fast_run.py)

**Comment by nevdelap** [comment]
I've added a couple of tests. Looking at how and where to put any tests in `test_fail_fast_run.py`.

**Comment by nevdelap** [comment]
Will it run those test suites and build packages eventually, or does it not do that until it gets a review?

**Comment by nevdelap** [comment]
I see Test Log Schema tests failed because they're doing something slightly different than running the test by itself does, so I'll look at that.

The Integration Tests against the different Python versions on different platforms failed for dependency version differences that are unrelated to these changes.

File: tests/functional/minimal_cli/test_minimal_cli.py [hunk 1]
```diff
@@ -53,6 +53,38 @@ def test_build(self, runner, project):
...
+        # 1 seed, 1 model, 2 data tests
+        assert "PASS=4" in result.output
+        # 2 data tests
+        assert "ERROR=2" in result.output
+        # Singular test
+        assert "WARN=1" in result.output
+        # 1 snapshot
+        assert "SKIP=1" in result.output
+        # Skipping due to fail_fast is shown when --debug is specified.
+        assert result.output.count("Skipping due to fail_fast") == 1
```
Comment (nevdelap) [inline tests/functional/minimal_cli/test_minimal_cli.py:85]: I'll change this to a plain 'in' test.

**Review by QMalcolm — APPROVED** [review]
Thanks for doing this work, it looks great 🙂

**Comment by scottgigante** [comment]
Thanks @nevdelap for picking up where I left off!

**Comment by scottgigante** [comment]
Closes #9371

**Comment by QMalcolm** [comment]
Our [codecov check](https://github.com/dbt-labs/dbt-core/pull/10244/checks?check_run_id=25915832174) is complaining. However, it's results don't really make sense. I'm gonna move forward with getting this merged.

**Comment by nevdelap** [comment]
Thanks @scottgigante, thanks @QMalcolm!

**Pull request merged**